ccm_web/server/src/invitation/cirrus-invitation.service.ts

```diff
@@ -0,0 +1,74 @@
+import axios from 'axios'
+import { randomUUID } from 'crypto'
+import FormData from 'form-data'
+import { lastValueFrom } from 'rxjs'
+import { HttpService } from '@nestjs/axios'
+import { ConfigService } from '@nestjs/config'
+import { Injectable } from '@nestjs/common'
+
+import { CirrusErrorData, CirrusInvitationResponse } from './cirrus-invitation.interfaces'
+import { InvitationAPIError } from './invitation.errors'
+
+import { Config } from '../config'
+import baseLogger from '../logger'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+@Injectable()
+export class CirrusInvitationService {
+  url: string
+  entityID: string
+  sponsorName: string
+  key: string
+  secret: string
+
+  constructor (
+    private readonly configService: ConfigService<Config, true>,
+    private readonly httpService: HttpService
+  ) {
+    const invitationConfig = configService.get('invitation', { infer: true })
+    this.url = invitationConfig.apiURL
+    this.entityID = invitationConfig.apiEntityID
+    this.sponsorName = invitationConfig.apiSponsorName
+    this.key = invitationConfig.apiKey
+    this.secret = invitationConfig.apiSecret
+  }
+
+  async sendInvitations (userEmails: string[]): Promise<CirrusInvitationResponse | CirrusErrorData> {
+    if (userEmails.length === 0) {
+      throw new InvitationAPIError('Argument "userEmails" array is empty.')
+    }
+
+    const emailAddressCSV = `emailAddress\n${userEmails.join('\n')}`
+
+    const data = new FormData()
+    data.append('cfile', emailAddressCSV, 'fake_file_name.csv')
```

Comment:
What's with `"fake_file_name.csv"`? Is this required? If so, can we use something a bit less "fake" looking?

Reply:
AFAIK, in order for a file attachment upload to work, it must be named.  It's not a requirement of Cirrus, that I know of, but the upload doesn't work without a name.  Maybe it's a `FormData` requirement.

Maybe what I should do is name it something similar to `clientRequestID`.

Reply:
AFAIK, in order for a file attachment upload to work, it must be named.  It's not a requirement of Cirrus, that I know of, but the upload doesn't work without a name.  Maybe it's a `FormData` requirement.

Maybe what I should do is name it something similar to `clientRequestID`.

Reply:
🏁